gst/datarepo/gstdatareposrc.c

```diff
@@ -1098,12 +1100,51 @@ gst_data_repo_src_start (GstDataRepoSrc * src)
   return FALSE;
 }
 
+/**
+ * @brief Set presentation timestamp and decoding timestamp
+ */
+static void
+gst_data_repo_src_set_pts_dts (GstDataRepoSrc * src, GstBuffer * buffer)
+{
+  GstClockTime next_time;
+
+  g_return_if_fail (src != NULL);
+  g_return_if_fail (buffer != NULL);
+
+  /* Unlike video, audio should control one sample by framerate. */
+  if (src->data_type == GST_DATA_REPO_DATA_AUDIO) {
+    GST_WARNING_OBJECT (src,
+        "Use audiorate element for the framerate of audio");
+    return;
+  }
+
+  GST_BUFFER_PTS (buffer) = src->running_time;
+  GST_BUFFER_DTS (buffer) = src->running_time;
+  gst_object_sync_values (GST_OBJECT (src), GST_BUFFER_PTS (buffer));
+
+  GST_BUFFER_OFFSET (buffer) = src->n_frame++;
```

Comment:
Q: why does the buffer offset need update?

Reply:
If don't update the buffer offset, fps doesn't work, so I updated it.

Reply:
I confused that, generally the buffer offset means the start of data in gst-buffer.

Reply:
I do not know well but
In videotestsrc, it seems to be used as the number of frames.

Reply:
Can you check adding do-timestamp in _src_init() function so that the buffer has timestamp?
`gst_base_src_set_do_timestamp (GST_BASE_SRC (src), TRUE);`

Reply:
okay, I will do that and share the result.